packages/agreement-process/src/model/domain/agreement-validators.ts

```diff
@@ -137,89 +139,94 @@ export const assertRequesterIsConsumer = (
   }
 };
 
-export function assertRequesterIsProducer(
+const assertRequesterIsProducer = (
   agreement: Agreement,
   authData: AuthData
-): void {
+): void => {
   if (
     !authData.userRoles.includes("internal") &&
     authData.organizationId !== agreement.producerId
   ) {
     throw operationNotAllowed(authData.organizationId);
   }
-}
+};
 
-export const assertRequesterIsConsumerOrProducer = (
+export const assertRequesterCanActAsConsumerOrProducer = (
   agreement: Agreement,
-  authData: AuthData
+  authData: AuthData,
+  activeProducerDelegation: Delegation | undefined
 ): void => {
   try {
     assertRequesterIsConsumer(agreement, authData);
   } catch (error) {
-    assertRequesterIsProducer(agreement, authData);
+    assertRequesterCanActAsProducer(
+      agreement,
+      authData,
+      activeProducerDelegation
+    );
   }
 };
 
-export const assertRequesterIsConsumerOrProducerOrDelegateProducer = async (
+export const assertRequesterCanRetrieveConsumerDocuments = async (
   agreement: Agreement,
   authData: AuthData,
   readModelService: ReadModelService
 ): Promise<void> => {
+  // This operation has a dedicated assertion because it's the only operation that
+  // can be performed also by the producer even when there is an active producer delegation
   try {
     assertRequesterIsConsumer(agreement, authData);
   } catch (error) {
     try {
       assertRequesterIsProducer(agreement, authData);
     } catch (error) {
-      const producerDelegation =
-        await readModelService.getDelegationByDelegateId(
-          authData.organizationId,
-          delegationKind.delegatedProducer
+      const activeProducerDelegation =
+        await readModelService.getActiveProducerDelegationByEserviceId(
+          agreement.eserviceId
         );
-      assertRequesterIsDelegate(producerDelegation?.delegateId, authData);
+      assertRequesterIsDelegateProducer(
+        agreement,
+        authData,
+        activeProducerDelegation
+      );
     }
   }
 };
 
-export const assertRequesterIsProducerOrDelegateProducer = (
+export const assertRequesterCanActAsProducer = (
```

Comment:
```suggestion
export const assertRequesterCanPerformProducerAction = (
```

Reply:
I would leave it as it is, "To act" is a shorter way to say "To perform an action", we would not add information with this rename